File: com.bosch.fsp.master_of_plow.gen/src/com/bosch/nevonex/fcal/impl/RECEIVE_ACC_INFO_T_R.java
/**
Copyright (c) Robert Bosch GmbH. All rights reserved.
*/
package com.bosch.nevonex.fcal.impl;

import com.bosch.nevonex.fcal.IRECEIVE_ACC_INFO_T_R;

import org.eclipse.emf.ecore.EClass;
import org.eclipse.emf.ecore.EObject;
import org.eclipse.emf.ecore.InternalEObject;

import org.eclipse.emf.ecore.impl.EObjectImpl;

/**
 * @generated
 */
public class RECEIVE_ACC_INFO_T_R extends EObjectImpl implements IRECEIVE_ACC_INFO_T_R {
	protected float tRZ_ACC_SIG1_V = 0.0f;
	protected float tRZ_ACC_SIG2_V = 0.0f;
	protected int tRZ_ACC_DIAG = 0;
	protected int tRZ_ACC_AUTO = 0;

	protected RECEIVE_ACC_INFO_T_R() {
		super();
	}

	@Override
	protected EClass eStaticClass() {
		return FcalPackage.Literals.RECEIVE_ACC_INFO_T_R;
	}

	public float getTRZ_ACC_SIG1_V() {
		return tRZ_ACC_SIG1_V;
	}

	public void setTRZ_ACC_SIG1_V(float newTRZ_ACC_SIG1_V) {
		tRZ_ACC_SIG1_V = newTRZ_ACC_SIG1_V;
	}

	public float getTRZ_ACC_SIG2_V() {
		return tRZ_ACC_SIG2_V;
	}

	public void setTRZ_ACC_SIG2_V(float newTRZ_ACC_SIG2_V) {
		tRZ_ACC_SIG2_V = newTRZ_ACC_SIG2_V;
	}

	public int getTRZ_ACC_DIAG() {
		return tRZ_ACC_DIAG;
	}

	public void setTRZ_ACC_DIAG(int newTRZ_ACC_DIAG) {
		tRZ_ACC_DIAG = newTRZ_ACC_DIAG;
	}

	public int getTRZ_ACC_AUTO() {
		return tRZ_ACC_AUTO;
	}

	public void setTRZ_ACC_AUTO(int newTRZ_ACC_AUTO) {
		tRZ_ACC_AUTO = newTRZ_ACC_AUTO;
	}

	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
		case FcalPackage.RECEIVE_ACC_INFO_T_R__TRZ_ACC_SIG1_V:
			return getTRZ_ACC_SIG1_V();
		case FcalPackage.RECEIVE_ACC_INFO_T_R__TRZ_ACC_SIG2_V:
			return getTRZ_ACC_SIG2_V();
		case FcalPackage.RECEIVE_ACC_INFO_T_R__TRZ_ACC_DIAG:
			return getTRZ_ACC_DIAG();
		case FcalPackage.RECEIVE_ACC_INFO_T_R__TRZ_ACC_AUTO:
			return getTRZ_ACC_AUTO();
		}
		return super.eGet(featureID, resolve, coreType);
	}

	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
		case FcalPackage.RECEIVE_ACC_INFO_T_R__TRZ_ACC_SIG1_V:
			setTRZ_ACC_SIG1_V((Float) newValue);
			return;
		case FcalPackage.RECEIVE_ACC_INFO_T_R__TRZ_ACC_SIG2_V:
			setTRZ_ACC_SIG2_V((Float) newValue);
			return;
		case FcalPackage.RECEIVE_ACC_INFO_T_R__TRZ_ACC_DIAG:
			setTRZ_ACC_DIAG((Integer) newValue);
			return;
		case FcalPackage.RECEIVE_ACC_INFO_T_R__TRZ_ACC_AUTO:
			setTRZ_ACC_AUTO((Integer) newValue);
			return;
		}
		super.eSet(featureID, newValue);
	}

	@Override
	public void eUnset(int featureID) {
		switch (featureID) {
		case FcalPackage.RECEIVE_ACC_INFO_T_R__TRZ_ACC_SIG1_V:
			setTRZ_ACC_SIG1_V(0.0f);
			return;
		case FcalPackage.RECEIVE_ACC_INFO_T_R__TRZ_ACC_SIG2_V:
			setTRZ_ACC_SIG2_V(0.0f);
			return;
		case FcalPackage.RECEIVE_ACC_INFO_T_R__TRZ_ACC_DIAG:
			setTRZ_ACC_DIAG(0);
			return;
		case FcalPackage.RECEIVE_ACC_INFO_T_R__TRZ_ACC_AUTO:
			setTRZ_ACC_AUTO(0);
			return;
		}
		super.eUnset(featureID);
	}

	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
		case FcalPackage.RECEIVE_ACC_INFO_T_R__TRZ_ACC_SIG1_V:
			return tRZ_ACC_SIG1_V != 0.0f;
		case FcalPackage.RECEIVE_ACC_INFO_T_R__TRZ_ACC_SIG2_V:
			return tRZ_ACC_SIG2_V != 0.0f;
		case FcalPackage.RECEIVE_ACC_INFO_T_R__TRZ_ACC_DIAG:
			return tRZ_ACC_DIAG != 0;
		case FcalPackage.RECEIVE_ACC_INFO_T_R__TRZ_ACC_AUTO:
			return tRZ_ACC_AUTO != 0;
		}
		return super.eIsSet(featureID);
	}
} //RECEIVE_ACC_INFO_T_R

File: master_of_plow_CPP_SDK/src-gen/nevonex/fcal/RECEIVE_ACC_INFO_T_R.cpp
/*
 * nevonex/fcal/RECEIVE_ACC_INFO_T_R.cpp
 * Copyright (c) Robert Bosch GmbH. All rights reserved.
 */

#include "RECEIVE_ACC_INFO_T_R.hpp"
#include <ecore/EObject.hpp>
#include <ecore/EClass.hpp>
#include "nevonex/fcal/FcalPackage.hpp"
#include <ecorecpp/mapping.hpp>
#include <algorithm>
#include <nevonex/fcb/SubscriberEnum.hpp>
#include <nevonex-fcal-platform/log/Logger.hpp>

#if BOOST_VERSION >= 106501
#define BOOST_STACKTRACE_HEADERS_FOUND
#include <boost/stacktrace.hpp>
#endif

/*PROTECTED REGION ID(RECEIVE_ACC_INFO_T_R.cpp) START*/
// Please, enable the protected region if you add manually written code.
// To do this, add the keyword ENABLED before START.
/*PROTECTED REGION END*/

using namespace ::nevonex::fcal;
using namespace ::nevonex::log;

// Default constructor
RECEIVE_ACC_INFO_T_R::RECEIVE_ACC_INFO_T_R() : m_tRZ_ACC_SIG1_V(0.0f), m_tRZ_ACC_SIG2_V(0.0f), m_tRZ_ACC_DIAG(0), m_tRZ_ACC_AUTO(0)
{

    /*PROTECTED REGION ID(RECEIVE_ACC_INFO_T_R__RECEIVE_ACC_INFO_T_R) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

RECEIVE_ACC_INFO_T_R::~RECEIVE_ACC_INFO_T_R()
{

    /*PROTECTED REGION ID(RECEIVE_ACC_INFO_T_R__Destructor) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

// Attributes

::ecore::EFloat RECEIVE_ACC_INFO_T_R::getTRZ_ACC_SIG1_V() const
{
    return m_tRZ_ACC_SIG1_V;
}

void RECEIVE_ACC_INFO_T_R::setTRZ_ACC_SIG1_V(::ecore::EFloat _tRZ_ACC_SIG1_V)
{

    ::ecore::EFloat _old_tRZ_ACC_SIG1_V = m_tRZ_ACC_SIG1_V;
    m_tRZ_ACC_SIG1_V = _tRZ_ACC_SIG1_V;

}

::ecore::EFloat RECEIVE_ACC_INFO_T_R::getTRZ_ACC_SIG2_V() const
{
    return m_tRZ_ACC_SIG2_V;
}

void RECEIVE_ACC_INFO_T_R::setTRZ_ACC_SIG2_V(::ecore::EFloat _tRZ_ACC_SIG2_V)
{

    ::ecore::EFloat _old_tRZ_ACC_SIG2_V = m_tRZ_ACC_SIG2_V;
    m_tRZ_ACC_SIG2_V = _tRZ_ACC_SIG2_V;

}

::ecore::EInt RECEIVE_ACC_INFO_T_R::getTRZ_ACC_DIAG() const
{
    return m_tRZ_ACC_DIAG;
}

void RECEIVE_ACC_INFO_T_R::setTRZ_ACC_DIAG(::ecore::EInt _tRZ_ACC_DIAG)
{

    ::ecore::EInt _old_tRZ_ACC_DIAG = m_tRZ_ACC_DIAG;
    m_tRZ_ACC_DIAG = _tRZ_ACC_DIAG;

}

::ecore::EInt RECEIVE_ACC_INFO_T_R::getTRZ_ACC_AUTO() const
{
    return m_tRZ_ACC_AUTO;
}

void RECEIVE_ACC_INFO_T_R::setTRZ_ACC_AUTO(::ecore::EInt _tRZ_ACC_AUTO)
{

    ::ecore::EInt _old_tRZ_ACC_AUTO = m_tRZ_ACC_AUTO;
    m_tRZ_ACC_AUTO = _tRZ_ACC_AUTO;

}

// References

